Sources/Apodini/Semantic Model Builder/GraphQL/GraphQLInterfaceExporter.swift

```diff
@@ -0,0 +1,150 @@
+                                              const graphQLFetcher = graphQLParams =>
+                                                fetch('/graphql', {
+                                                  method: 'post',
+                                                  headers: { 'Content-Type': 'application/json' },
+                                                  body: JSON.stringify(graphQLParams),
+                                                })
+                                                  .then(response => response.json())
+                                                  .catch(() => response.text());
+                                              ReactDOM.render(
+                                                React.createElement(GraphiQL, { fetcher: graphQLFetcher }),
+                                                document.getElementById('graphiql'),
+                                              );
+                                            </script>
+                                          </body>
+                                        </html>
+                                        """
+
+        return Vapor.Response(status: .ok, headers: ["Content-Type": "text/html"], body: html)
+    }
+
+    private func graphqlServer(_ req: Vapor.Request) throws -> EventLoopFuture<String> {
+        guard let body = req.body.string else {
+            throw NetworkError.noBody
+        }
+
+        // Create Swift dict
+        var query = "{}"
+        let data = body.data(using: .utf8)!
+        do {
+            if let jsonArray = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as? [String: AnyObject] {
+                if let q = jsonArray["query"] as? String {
+                    query = q
+                } else {
+                    print("jsonArray doesn't have query parameter")
+                }
+
+            } else {
+                print("bad json")
+                throw NetworkError.noBody
+            }
+        } catch let error as NSError {
+            print(error)
+            throw NetworkError.noBody
+        }
+
+        return try graphql(schema: self.schema,
+                request: query,
+                context: req,
+                eventLoopGroup: req.eventLoop).map { result -> String in
+            result.description
+        }
+
+    }
+
+    // We should have parameter for the result struct values and paramters
+    func export<H: Handler>(_ endpoint: Endpoint<H>) {
+        self.graphQLPath.append(for: endpoint, with: endpoint.createConnectionContext(for: self))
+    }
+
+    func exportParameter<Type: Codable>(_ parameter: EndpointParameter<Type>) -> String {
+        parameter.name
+    }
+
+    func finishedExporting(_ webService: WebServiceModel) {
+        self.schema = self.graphQLPath.generate()
+    }
+
+    func retrieveParameter<Type: Decodable>(_ parameter: EndpointParameter<Type>, for request: GraphQLRequest) throws -> Type?? {
```

Comment:
I think that the current implementation is missing a lot of different edge cases that must be covered here. E.g. complex structs or elements can not be decoded in the moment. I think it would be interesting how the decoding for more complex types would work.
I think we will need to collect what parameters are possible for each endpoint or need to extend the `InterfaceExporter` to get all Parameters at once and return a value for each one step by step. E.g. it make a difference if there is only one parameter (e.g. one `Int`) and therefore we can use the top level value or of we have two parameters (`name: String`, `age: Int`) and therefore need to search for a dictionary that contains String keys and different ambiguous values that can be decoded for the corresponding keys. As far as I can see this is currently not done by any other exporter but #131 is approaching this for the Protocol Buffer generator.